tools/trace_analyzer_tool.cc

```diff
@@ -469,96 +472,29 @@ Status TraceAnalyzer::StartProcessing() {
       break;
     }
 
+    // To do: why increment total_requests_ for trace end?
```

Comment:
Sometimes, the trace might be incomplete. We may not be able to the type::end.

Reply:
Yes, my point is, you should not increment total_requests if you read a `TraceEnd`, so it should be 
```
    end_time_ = trace.ts;
    if (trace.type == kTraceEnd) {
      break;
    }
    total_requests_++;
```
Then `end_time_` is the timetamp of the last trace (or trace end), and `total_requests_` will be the actual number of requests.

Reply:
> Yes, my point is, you should not increment total_requests if you read a `TraceEnd`, so it should be
> 
> ```
>     end_time_ = trace.ts;
>     if (trace.type == kTraceEnd) {
>       break;
>     }
>     total_requests_++;
> ```
> 
> Then `end_time_` is the timetamp of the last trace (or trace end), and `total_requests_` will be the actual number of requests.

Sure, just follow what you proposed above.

Reply:
Done